Code/Mantid/Vates/VatesAPI/src/MDHWNexusLoadingPresenter.cpp

```diff
@@ -151,5 +127,57 @@ std::string MDHWNexusLoadingPresenter::getWorkspaceTypeName()
   return m_wsTypeName;
 }
 
+std::vector<int> MDHWNexusLoadingPresenter::getExtents()
+{
+
+  std::vector<int> extents(6, 0);
+
+  if(this->shouldLoad() || !m_histoWs)
+    this->loadWorkspace();
+
+  if(m_histoWs)
+  {
+    extents[1] = static_cast<int>(m_histoWs->getXDimension()->getNBins());
+    extents[3] = static_cast<int>(m_histoWs->getYDimension()->getNBins());
+    extents[5] = static_cast<int>(m_histoWs->getZDimension()->getNBins());
+  }
+
+  return extents;
+
+}
+
+void MDHWNexusLoadingPresenter::loadWorkspace()
+{
+  using namespace Mantid::API;
+  AnalysisDataService::Instance().remove("MD_HISTO_WS_ID");
+
+  IAlgorithm_sptr alg = AlgorithmManager::Instance().create("LoadMD");
+  alg->initialize();
+  alg->setPropertyValue("Filename", this->m_filename);
+  alg->setPropertyValue("OutputWorkspace", "MD_HISTO_WS_ID");
+  alg->setProperty("FileBackEnd", !this->m_view->getLoadInMemory()); //Load from file by default.
+  alg->execute();
+  Workspace_sptr result = AnalysisDataService::Instance().retrieve("MD_HISTO_WS_ID");
+  m_histoWs = boost::dynamic_pointer_cast<Mantid::API::IMDHistoWorkspace>(result);
+}
+
+void MDHWNexusLoadingPresenter::loadWorkspace( ProgressAction& loadingProgressUpdate)
+{
+    using namespace Mantid::API;
+    Poco::NObserver<ProgressAction, Mantid::API::Algorithm::ProgressNotification> observer(loadingProgressUpdate, &ProgressAction::handler);
+
+    AnalysisDataService::Instance().remove("MD_HISTO_WS_ID");
+    IAlgorithm_sptr alg = AlgorithmManager::Instance().create("LoadMD");
+    alg->initialize();
+    alg->setPropertyValue("Filename", this->m_filename);
+    alg->setPropertyValue("OutputWorkspace", "MD_HISTO_WS_ID");
+    alg->setProperty("FileBackEnd", !this->m_view->getLoadInMemory()); //Load from file by default.
+    alg->addObserver(observer);
+    alg->execute();
+    alg->removeObserver(observer);
+    Workspace_sptr result = AnalysisDataService::Instance().retrieve("MD_HISTO_WS_ID");
+    m_histoWs = boost::dynamic_pointer_cast<Mantid::API::IMDHistoWorkspace>(result);
```

Comment:
Just a comment about these last two lines. There's this alternative way:
`m_histoWS = AnalysisDataService::Instance().retrieveWS<Mantid::API::IMDHistoWorkspace>("name");`
which would be equivalent. I tend to prefer it as it is just one line and you use a more specific call, but I'm not really sure if there's any general recommendation and/or preference.